src/components/app/BeforeUnloadManager.js

```diff
@@ -0,0 +1,56 @@
+/* this source code form is subject to the terms of the mozilla public
+ * license, v. 2.0. if a copy of the mpl was not distributed with this
+ * file, you can obtain one at http://mozilla.org/mpl/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import explicitConnect from '../../utils/connect';
+
+import type { ConnectedProps } from '../../utils/connect';
+import { getUploadPhase } from '../../selectors/publish';
+
+type StateProps = {|
+  +isUploading: boolean,
+|};
+
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+class BeforeUnloadManager extends React.PureComponent<Props> {
+  manageBeforeUnloadListener() {
+    const { isUploading } = this.props;
+    if (isUploading) {
+      window.addEventListener('beforeunload', this.handleUnload);
+    } else {
+      window.removeEventListener('beforeunload', this.handleUnload);
+    }
+  }
+
+  handleUnload = (event: BeforeUnloadEvent) => {
+    event.preventDefault();
+    return 'Are you sure you want to close while uploading?';
```

Comment:
In Firefox everything works fine, but in Chrome I don't have it.
Maybe returning a string doesn't work after all ([that's what MDN was saying though](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event#Browser_compatibility)). The HTML spec doesn't say anything about that either actually (see step 8 in [this part of the html spec](https://html.spec.whatwg.org/#unloading-documents)).

Soooo I would suggest to do `event.returnValue = XXX` instead of returning it.
Sorry about that.

Will you be also kind enough to update the test to check for that `returnValue` property?

Reply:
Sure thing. `fireEvent` still seems to return a boolean, so I left the check for return value in the test unchanged. I tried it on chrome and it seems to be working there with the new change. I read the spec that you linked and it seems like chrome does not support `preventDefault()` but it does support changing returnValue.